File: samples/AdventureWorks/AdventureWorks.Client.Common/DataObjects/Sales/SalesOrderCriteriaCustomized.cs
using AdventureWorks.Services;
using System;
using System.Threading;
using Xomega.Framework;

namespace AdventureWorks.Client.Objects
{
    public class SalesOrderCriteriaCustomized : SalesOrderCriteria
    {
        public SalesOrderCriteriaCustomized()
        {
        }

        public SalesOrderCriteriaCustomized(IServiceProvider serviceProvider) : base(serviceProvider)
        {
        }

        // construct properties and child objects
        protected override void Initialize()
        {
            base.Initialize();
            // add custom construction code here
        }

        // perform post intialization
        protected override void OnInitialized()
        {
            base.OnInitialized();
            StatusProperty.DisplayFormat = Header.FieldId + " - " + Header.FieldText;
            TerritoryIdProperty.SetCascadingProperty(Enumerations.SalesTerritory.Attributes.Group, GlobalRegionProperty);
            SalesPersonIdProperty.SetCascadingProperty(Enumerations.SalesPerson.Attributes.TerritoryId, TerritoryIdProperty);

            if (Thread.CurrentPrincipal.IsStoreContact() || Thread.CurrentPrincipal.IsIndividualCustomer())
            {
                CustomerStoreOperatorProperty.AccessLevel = AccessLevel.None;
                CustomerNameOperatorProperty.AccessLevel = AccessLevel.None;
            }
        }

        public override void Validate(bool force)
        {
            base.Validate(force);
            DateTime? orderDateFrom = OrderDateProperty.Value;
            DateTime? orderDateTo = OrderDate2Property.Value;
            if (orderDateFrom != null && orderDateTo != null && orderDateTo < orderDateFrom)
                validationErrorList.AddError(ErrorType.Validation, "From Order Date should be earlier than To Order Date");
        }
    }
}

File: samples/AdventureWorks/AdventureWorks.Client.Spa/DataObjects/Sales/SalesOrderCriteriaCustomized.ts
import { SalesOrderCriteria as GeneratedDataObject } from 'DataObjects/Sales/SalesOrderCriteria';
import { PersonType, SalesTerritory } from 'Enumerations/Enumerations'
import { AuthManager, AccessLevel, Header } from 'xomega'

export class SalesOrderCriteria extends GeneratedDataObject {

    // perform post intialization
    onInitialized() {
        super.onInitialized();
        this.Status.DisplayFormat = Header.fieldId + ' - ' + Header.fieldText;
        this.TerritoryId.setCascadingProperty(SalesTerritory.Attributes.Group, this.GlobalRegion);

        let claims = AuthManager.Current.Claims;
        if (claims && (claims.role == PersonType.StoreContact
                    || claims.role == PersonType.IndividualCustomer)) {
            this.CustomerStoreOperator.AccessLevel(AccessLevel.None);
            this.CustomerNameOperator.AccessLevel(AccessLevel.None);
        }
    }

    public validate(force: boolean) {
        super.validate(force);
        if (!this.OrderDate.isNull() && !this.OrderDate2.isNull() &&
            this.OrderDate.InternalValue() > this.OrderDate2.InternalValue())
            this.ValidationErrors.addError('From Order Date should be earlier than To Order Date');
    }
}